core/src/main/java/org/hisp/dhis/android/core/analytics/aggregated/DimensionalResponse.kt

```diff
@@ -38,5 +40,6 @@ data class DimensionalResponse(
 
 data class DimensionalValue(
     val dimensions: List<String>,
-    val value: String?
+    val value: String?,
+    val legend: Legend?
```

Comment:
An alternative option would be to have `val legend: String?`, so the legend is not a whole Legend object but just the uid of the Legend. Then, the whole Legend object will be included in the metadata map. This is the strategy followed for `val dimensions: List<String>`, they only contain a list of uids and the metadata map contains the whole objects (name, description, etc). The purpose of this is to reduce the size of the response and avoid having duplicated objects.
I am not sure that this strategy makes sense for Legends becuase it can be very handy to have whole Legend object (including the color) when printing the value in the table. You have implemented this in the App side, so you know better than me. What do you think about this?

Reply:
@vgarciabnz sorry, we understood this was actually what you wanted us to do (misunderstanding probably from our side). In any case, this is also implemented like this in TEI analytics and its modification would (as you properly stated) also imply modifying the app code. Would you mind if we do this as a refactor after this PR, so this can be validated as-is for this release and next one contains the refactor? That would give us the possibility of checking how's the performance impact before moving the code to a new version without duplicated objects

Reply:
I agree on merging the current PR and look at this later. Actually, I am not completely sure about replacing the Legend by just the uid: the benefit of having the whole Legend object when printing the values could be higher that the difference in performance. Just wanted to know if you had considered that option.